pkg/api/v20230701preview/openshiftcluster_validatestatic_test.go

```diff
@@ -517,8 +522,171 @@ func TestOpenShiftClusterStaticValidateNetworkProfile(t *testing.T) {
 			},
 			wantErr: "",
 		},
+		{
+			name: "LoadbalancerProfile invalid when used with UserDefinedRouting",
+			current: func(oc *OpenShiftCluster) {
+				oc.Properties.NetworkProfile.OutboundType = OutboundTypeUserDefinedRouting
+				oc.Properties.IngressProfiles[0].Visibility = VisibilityPrivate
+				oc.Properties.APIServerProfile.Visibility = VisibilityPrivate
+				oc.Properties.NetworkProfile.LoadbalancerProfile = &LoadbalancerProfile{
+					ManagedOutboundIPs: &ManagedOutboundIPs{
+						Count: 3,
+					},
+				}
+			},
+			wantErr: "400: InvalidParameter: properties.networkProfile.loadBalancerProfile: The provided loadBalancerProfile is invalid: cannot use a loadBalancerProfile if outboundType is UserDefinedRouting.",
+		},
+	}
+
+	runTests(t, testModeCreate, tests)
+	runTests(t, testModeUpdate, tests)
+}
+
+func TestOpenShiftClusterStaticValidateLoadBalancerProfile(t *testing.T) {
+	tests := []*validateTest{
+		{
+			name:    "LoadbalancerProfile is valid",
+			wantErr: "",
+		},
+		{
+			name: "LoadbalancerProfile is invalid with ManagedOutboundIPs.Count and OutboundIPs set",
+			current: func(oc *OpenShiftCluster) {
+				oc.Properties.NetworkProfile.LoadbalancerProfile = &LoadbalancerProfile{
+					ManagedOutboundIPs: &ManagedOutboundIPs{
+						Count: 3,
+					},
+					OutboundIPs: []OutboundIP{
+						{
+							ID: "outboundIPID",
+						},
+					},
+					OutboundIPPrefixes: nil,
+				}
+			},
+			wantErr: "400: InvalidParameter: properties.networkProfile.loadBalancerProfile: The provided loadBalancerProfile is invalid: can only use one of managedOutboundIps, outboundIps, or outboundIpPrefixes at a time.",
+		},
+		{
+			name: "LoadbalancerProfile is invalid with all fields nil",
+			current: func(oc *OpenShiftCluster) {
+				oc.Properties.NetworkProfile.LoadbalancerProfile = &LoadbalancerProfile{
+					ManagedOutboundIPs: nil,
+					OutboundIPs:        nil,
+					OutboundIPPrefixes: nil,
+				}
+			},
+			wantErr: "400: InvalidParameter: properties.networkProfile.loadBalancerProfile: The provided loadBalancerProfile is invalid: must specify one of managedOutboundIps, outboundIps, or outboundIpPrefixes.",
+		},
+		{
+			name: "LoadbalancerProfile is invalid with OutboundIPs and OutboundIPPrefixes set",
+			current: func(oc *OpenShiftCluster) {
+				oc.Properties.NetworkProfile.LoadbalancerProfile = &LoadbalancerProfile{
+					ManagedOutboundIPs: nil,
+					OutboundIPs: []OutboundIP{
+						{
+							ID: "outboundIPID",
+						},
+					},
+					OutboundIPPrefixes: []OutboundIPPrefix{
+						{
+							ID: "outboundIPPrefixID",
+						},
+					},
+				}
+			},
```

Comment:
I know it's gonna be a lot of copy-pasta, but I'd appreciate having all permutations of this in our tests to give us assurance. No combination of any of these fields should be allowed. I guess I'm assuming that this error would precede the "this field is unsupported" errors.

Reply:
If more than 1 of these fields is specified at a time will the `else` will catch it and return `"The provided loadBalancerProfile is invalid: can only use one of managedOutboundIps, outboundIps, or outboundIpPrefixes at a time."`.  If a single field is specified (which is correct as these 3 fields are mutually exclusive) but the field is not yet supported it will return "not implemented".  

I will add the additional permutations.